R/sensitivity_labels.R

```diff
@@ -40,23 +41,66 @@ read_sensitivity_label <- function(file) {
     )
   }
 
-  # Check file is Excel workbook
-  file_extension <- tolower(tools::file_ext(file))
-  if (!file_extension %in% c("xlsx", "xls")) {
+  # Check file is valid
+  file_ext <- tolower(tools::file_ext(file))
+  if (!file_ext %in% c("xlsx", "xls", "docx")) {
     cli::cli_abort(
-      "{.arg file} must be an Excel workbook with {.val .xlsx} or {.val .xls} extension, not {.val .{file_extension}}."
+      "{.arg file} must be an Excel workbook or Word document with {.val .xlsx}, {.val .xls}, or {.val .docx} extension, not {.val .{file_ext}}."
     )
   }
 
-  wb <- openxlsx2::wb_load(file)
-  mips <- openxlsx2::wb_get_mips(wb)
+  ## Extracting label within Excel workbooks ----
 
-  if (is.null(mips) || length(mips) == 0L || mips == "") {
-    return("No label")
+  if (file_ext %in% c("xlsx", "xls")) {
+    wb <- openxlsx2::wb_load(file)
+    mips <- openxlsx2::wb_get_mips(wb)
+
+    if (is.null(mips) || length(mips) == 0L || mips == "") {
+      return("No label")
+    }
+
+    # Try to extract label name from XML
+    label_name <- which(sensitivity_label_xml == mips) |> names()
   }
 
-  # Try to extract label name from XML
-  label_name <- which(sensitivity_label_xml == mips) |> names()
+  ## Extracting label within Word docs ----
+
+  if (file_ext == "docx") {
+    file_name <- basename(file)
+    file_name <- tools::file_path_sans_ext(file_name)
+
+    zipdir <- file.path(tempdir(), file_name)
+
+    if (grepl("ming.32", R.Version()$platform)) {
+      zipdir <- gsub("\\\\", "/", zipdir)
+    }
```

Comment:
Are we sure this is needed? If I do `file.path(tempdir(), "1234")` on RStudio Desktop (Windows), I get `"C:\\Users\\jamesm32\\AppData\\Local\\Temp\\RtmpygyES6/1234"` which looks horrible, but does work fine.

I usually use `{fs}` where `fs::path` does a similar platform agnostic thing with file paths, but they look prettier `C:/Users/jamesm32/AppData/Local/Temp/RtmpygyES6/1234`. I didn't suggest using `{fs}` as it would be a new dependency.
```suggestion
```